app/scripts/modules/core/src/artifact/expectedArtifact.service.ts

```diff
@@ -39,7 +38,7 @@ export class ExpectedArtifactService {
     // is that we don't want to include an empty string in the artifact list, so omit any falsey values.
     return (stageContext: IExecutionContext) =>
       fields
-        .map((field): string => get(stageContext, field))
```

Comment:
😮

Reply:
oh, actually, yes, this is bad - there is one place where there's a dot-nested reference so we should use `get` here